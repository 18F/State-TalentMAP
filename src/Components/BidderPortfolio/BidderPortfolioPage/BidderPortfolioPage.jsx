import React from 'react';
import PropTypes from 'prop-types';
import { BIDDER_LIST, BIDDER_PORTFOLIO_COUNTS } from '../../../Constants/PropTypes';
import Spinner from '../../Spinner';
import BidderPortfolioContainer from '../BidderPortfolioContainer';
import TopNav from '../TopNav';
import BidControls from '../BidControls';
import BidderPortfolioSearch from '../BidderPortfolioSearch';

const BidderPortfolioPage = ({ bidderPortfolio, bidderPortfolioIsLoading,
bidderPortfolioHasErrored, pageSize, queryParamUpdate, pageNumber,
bidderPortfolioCounts }) => {
  // here we just want to check that the all prop exists, because we want the nav section to appear
  // even when we reload the counts
  let navDataIsLoading = !bidderPortfolioCounts.all;
  // If we can determine that the all prop is zero, then we'll change navDataIsLoading to false,
  // since zero normally evaluates as false.
  if (bidderPortfolioCounts.all === 0) { navDataIsLoading = false; }
  // for bidder results, however, we'll wait until everything is loaded
  const isLoading = (bidderPortfolioIsLoading && !bidderPortfolioHasErrored) ||
    navDataIsLoading;
  return (
    <div className="bidder-portfolio-page">
      <BidderPortfolioSearch onUpdate={queryParamUpdate} />
      <div className="usa-grid-full bidder-portfolio-container profile-content-inner-container">
        {
          !navDataIsLoading &&
          <div>
            <TopNav bidderPortfolioCounts={bidderPortfolioCounts} />
            <BidControls
              queryParamUpdate={queryParamUpdate}
              biddersNumerator={bidderPortfolio.count}
              biddersDenominator={bidderPortfolioCounts.all}
              isLoading={isLoading}
            />
          </div>
        }
        <div className={`usa-grid-full bidder-portfolio-listing ${isLoading ? 'results-loading' : null}`}>
          {
            isLoading &&
              <Spinner type="homepage-position-results" size="big" />
          }
          {
            !isLoading &&
              <BidderPortfolioContainer
                bidderPortfolio={bidderPortfolio}
                pageSize={pageSize}
                queryParamUpdate={queryParamUpdate}
                pageNumber={pageNumber}
              />
          }
        </div>
      </div>
    </div>
  );
};

BidderPortfolioPage.propTypes = {
  bidderPortfolio: BIDDER_LIST.isRequired,
  bidderPortfolioIsLoading: PropTypes.bool.isRequired,
  bidderPortfolioHasErrored: PropTypes.bool.isRequired,
  pageSize: PropTypes.number.isRequired,
  queryParamUpdate: PropTypes.func.isRequired,
  pageNumber: PropTypes.number.isRequired,
  bidderPortfolioCounts: BIDDER_PORTFOLIO_COUNTS.isRequired,
};

export default BidderPortfolioPage;
