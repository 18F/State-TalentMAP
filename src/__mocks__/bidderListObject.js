const bidderListObject = {
  count: 3,
  results: [
    {
      id: 1,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'John',
        last_name: 'Doe',
        email: 'doej@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
    {
      id: 2,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'Jane',
        last_name: 'Doe',
        email: 'doejane@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
    {
      id: 3,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'John',
        last_name: 'Smith',
        email: 'smithj@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
  ],
};

export default bidderListObject;
