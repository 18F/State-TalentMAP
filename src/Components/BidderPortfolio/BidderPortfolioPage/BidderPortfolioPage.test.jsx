import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import BidderPortfolioPage from './BidderPortfolioPage';
import bidderListObject from '../../../__mocks__/bidderListObject';

describe('BidderPortfolioPageComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<BidderPortfolioPage
      bidderPortfolio={bidderListObject}
      bidderPortfolioIsLoading={false}
      bidderPortfolioHasErrored={false}
      pageSize={8}
      queryParamUpdate={() => {}}
      pageNumber={1}
    />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<BidderPortfolioPage
      bidderPortfolio={bidderListObject}
      bidderPortfolioIsLoading={false}
      bidderPortfolioHasErrored={false}
      pageSize={8}
      queryParamUpdate={() => {}}
      pageNumber={1}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
