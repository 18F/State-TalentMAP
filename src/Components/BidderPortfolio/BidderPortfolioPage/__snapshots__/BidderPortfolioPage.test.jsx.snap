// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidderPortfolioPageComponent matches snapshot 1`] = `
<div
  className="bidder-portfolio-page card-view"
>
  <BidderPortfolioSearch
    onUpdate={[Function]}
  />
  <div
    className="usa-grid-full bidder-portfolio-container profile-content-inner-container"
  >
    <div>
      <TopNav
        bidderPortfolioCounts={
          Object {
            "all": 100,
            "bidding": 50,
            "inpanel": 20,
            "inpost": 30,
          }
        }
      />
      <BidControls
        biddersDenominator={100}
        biddersNumerator={3}
        changeViewType={[Function]}
        isLoading={false}
        queryParamUpdate={[Function]}
        viewType="card"
      />
    </div>
    <div
      className="usa-grid-full bidder-portfolio-listing "
    >
      <BidderPortfolioContainer
        bidderPortfolio={
          Object {
            "count": 3,
            "results": Array [
              Object {
                "id": 1,
                "is_cdo": false,
                "skill_code": "INFORMATION MANAGEMENT (2880)",
                "user": Object {
                  "email": "doej@state.gov",
                  "first_name": "John",
                  "last_name": "Doe",
                },
              },
              Object {
                "id": 2,
                "is_cdo": false,
                "skill_code": "INFORMATION MANAGEMENT (2880)",
                "user": Object {
                  "email": "doejane@state.gov",
                  "first_name": "Jane",
                  "last_name": "Doe",
                },
              },
              Object {
                "id": 3,
                "is_cdo": false,
                "skill_code": "INFORMATION MANAGEMENT (2880)",
                "user": Object {
                  "email": "smithj@state.gov",
                  "first_name": "John",
                  "last_name": "Smith",
                },
              },
            ],
          }
        }
        pageNumber={1}
        pageSize={8}
        queryParamUpdate={[Function]}
        showListView={false}
      />
    </div>
  </div>
</div>
`;
